### PR TITLE
[tune] Fix testResourceScheduler and testMultiStepRun.

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/utils/mock.py [hunk 1]
```diff
@@ -155,3 +155,25 @@ def on_step_begin(self, **info):
...
+        new_snapshot = defaultdict(str)
+        for trial in trials:
+            new_snapshot[trial.trial_id] = trial.status
+        if not new_snapshot:
+            # Don't add an empty snapshot.
+            return
+        if not self._trial_status_snapshots or any(
+                new_snapshot[trial_id] != self._trial_status_snapshots[-1].get(
+                    trial_id, "") for trial_id in new_snapshot.keys()):
+            self._trial_status_snapshots.append(new_snapshot)
```
Comment (krfricke) [inline python/ray/tune/utils/mock.py:179]: Please add a comment here explaining the logic (it's hard to grasp from a glance what we're doing here).

Also consider just doing
```suggestion
        if not self._trial_status_snapshots or self._trial_status_snapshots[-1] != new_snapshot:
            self._trial_status_snapshots.append(new_snapshot)
```
as dict comparisons should be fine here

**Review by krfricke — APPROVED** [review]
Great, just a minor nit

**Comment by xwjiang2010** [comment]
I will take a look at other rllib originated Trainables in Tune tests separately after this is merged.

**Review by krfricke — APPROVED** [review]
Looks good!

**Pull request merged**